### ISSUE TITLE
day06 bash improve performance

### ISSUE THREAD

**Issue by Arxcis** [Arxcis/adventofcode2018#38]
Finishes in 8 minutes on my machine
```
  ✔ day06-chronal-coordinates › test › main.bash (8m 18.4s)
    ℹ 3909 === 3909 1 of 2
    ℹ 36238 === 36238 2 of 2
```

Would be interesting to investigate if something can be done to improve performance.

I am not qualified yet, as I have not completed the task in any other language. But I will be ready to discuss as soon as I get there.

**Comment by Arxcis** [comment]
`cpp` results are in = 2225x faster (more than 3 orders of magnitude faster)
```
$ g++ -std=c++17 main.cpp -O3 -o main && time cat input | ./main 
3909
36238

real	0m0.022s
user	0m0.020s
sys	0m0.003s
```

vs. bash
```
$ time cat input | bash main.bash 
3909
36238

real	7m25.914s
user	7m25.325s
sys	0m0.306s

```

**Comment by tholok97** [comment]
xD. Think logic needs to be fixed before performance can be prioritized #39. Looking into it soon.

**Comment by tholok97** [comment]
Version with fixed logic (PR #44) finishes in 2 minutes 40 seconds on my machine.

```
$ npm run generate && npm run bash day06-chronal-coordinates/

> adventofcode2018@ generate /home/tholok/code/adventofcode2018
> node generate.js

Generated day02-inventory-management-system/test.js
Generated day01-chronal-calibration/test.js
Generated day04-respose-record/test.js
Generated day05-alchemical-reduction/test.js
Generated day06-chronal-coordinates/test.js
Generated day07-the-sum-of-its-parts/test.js
Generated day09-marble-mania/test.js
Generated day10-the-stars-align/test.js
Generated day14-chocolates-chart/test.js
Generated day13-mine-cart-madness/test.js
Generated day15-beverage-bandits/test.js
Generated day11-chronal-charge/test.js
Generated day08-memory-maneuver/test.js
Generated day12-subterranean-sustainability/test.js
Generated day03-no-matter-how-you-slice-it/test.js

> adventofcode2018@ bash /home/tholok/code/adventofcode2018
> ava --verbose --match='*.bash' "day06-chronal-coordinates/"


  ✔ main.bash (2m 40.9s)
    ℹ 3909 === 3909 1 of 2
    ℹ 36238 === 36238 2 of 2

  1 test passed
```

Used @Arxcis C++ code as the template for the logic fix. Previous version used 4 minutes locally (as seen in original PR #34), so there is at least some improvement :yum: Curious to see how fast Travis can do it.

Strange that it takes 7 minutes on your machine btw!

**Comment by Arxcis** [comment]
Hmm. Well that is an improvement I can be satisified with 👍  Cool that you could use some of the cpp code as inspo.

**Comment by Arxcis** [comment]
Fixed by #44
